compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1220,6 +1220,34 @@ LogicalResult ArgCompareOp::reifyResultShapes(
       .reifyResultShapes(b, reifiedReturnShapes);
 }
 
+SmallVector<AffineMap>
+IREE::LinalgExt::ArgCompareOp::getIndexingMapsForOperands() {
+  Builder b(getContext());
+  const int64_t rank = getInputRank();
+  return SmallVector<AffineMap>{b.getMultiDimIdentityMap(rank)};
+}
+
+SmallVector<AffineMap>
+IREE::LinalgExt::ArgCompareOp::getIndexingMapsForResults() {
+  MLIRContext *ctx = getContext();
+  const int64_t rank = getInputRank();
+  const int64_t redDim = static_cast<int64_t>(getDimension());
+
+  SmallVector<AffineExpr> proj;
+  for (int64_t i = 0; i < rank; ++i) {
+    if (i == redDim)
+      continue;
+    proj.push_back(getAffineDimExpr(i, ctx));
+  }
+
+  AffineMap resultMap = AffineMap::get(rank, 0, proj, ctx);
+  return SmallVector<AffineMap>{resultMap, resultMap};
+}
+
+SmallVector<int64_t> IREE::LinalgExt::ArgCompareOp::getStaticLoopRanges() {
+  return SmallVector<int64_t>(getInputType().getShape());
```

Comment:
nit: you can use brace initialization
```suggestion
  return {getInputType().getShape()};
```

Reply:
`getInputType().getShape()` returns `ArrayRef<int64_t>`, SmallVector’s ArrayRef constructor is explicit, so return {getInputType().getShape()} can’t not be used here.

Reply:
But it can be used for other cases you pointed to, thanks.

Reply:
the alternative is to use `llvm::to_vector(someArrayRef)` -- you don't need the exact type. But fine to keep as-is.